### PR TITLE
Added ttl parameter to PutObjectHandler method

### PR DESCRIPTION
**Pull request by togettoyou**
# What problem are we solving?

We are addressing the need for controlling the expiration of objects in SeaweedFS. Currently, there is no mechanism to set a time-to-live (ttl) for uploaded objects.

# How are we solving the problem?

To solve this problem, I have added a new code snippet to the PutObjectHandler method in SeaweedFS. This code snippet extracts the ttl value from the "x-amz-meta-ttl" header and appends it as a query parameter to the upload URL. By doing this, users can now specify the ttl for each uploaded object.

# How is the PR tested?

The PR has been tested in the following way:

- Created a local development environment.
- Performed multiple upload operations with different ttl values.
- Verified that the ttl value is correctly extracted from the request header and appended to the upload URL.
- Ensured that the uploaded objects are expired based on the specified ttl.

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

**Comment by chrislusf** [comment]
Is this part of the S3 api?

**Comment by togettoyou** [comment]
AWS S3 does not natively support setting expiration time for files, this feature is specific to SeaweedFS S3. Here's an example of how you can use this feature:

```go
package main

import (
	"bytes"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/credentials"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3"
)

func main() {
	sess := session.Must(session.NewSession(&aws.Config{
		Credentials:      credentials.NewStaticCredentials("some_access_key1", "some_secret_key1", ""),
		Endpoint:         aws.String("127.0.0.1:8333"),
		Region:           aws.String("us-east-1"),
		DisableSSL:       aws.Bool(true),
		S3ForcePathStyle: aws.Bool(true),
	}))
	svc := s3.New(sess)
	inputObject := &s3.PutObjectInput{
		Bucket: aws.String("test"),
		Key:    aws.String("hello.txt"),
		Body:   bytes.NewReader([]byte("hello")),
	}
	ttl := "1m"
	inputObject.SetMetadata(map[string]*string{"ttl": &ttl})
	_, err := svc.PutObject(inputObject)
	if err != nil {
		panic(err)
	}
}
```

In this example, we're using the AWS SDK for Go to interface with SeaweedFS S3. We create a new PutObjectInput with the desired bucket, key, and object body. We also set the ttl metadata field to specify the expiration time for the object. By calling svc.PutObject, the object is uploaded to SeaweedFS S3 with the specified TTL.

**Comment by togettoyou** [comment]
Java S3 Usage Example:

```java
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.client.builder.AwsClientBuilder;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import com.amazonaws.services.s3.model.*;

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.util.HashMap;

public class S3TTLTest {

    public static void main(String[] args) {
        BasicAWSCredentials awsCredentials = new BasicAWSCredentials("some_access_key1", "some_secret_key1");
        AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("http://127.0.0.1:8333", "us-east-1")).withCredentials(new AWSStaticCredentialsProvider(awsCredentials)).build();

        String content = "hello";
        byte[] contentBytes = content.getBytes();
        InputStream inputStream = new ByteArrayInputStream(contentBytes);

        ObjectMetadata metadata = new ObjectMetadata();
        HashMap<String, String> userMeta = new HashMap<>();
        userMeta.put("ttl", "1m");
        metadata.setUserMetadata(userMeta);

        PutObjectRequest putRequest = new PutObjectRequest("test", "hello.txt", inputStream, metadata);
        s3Client.putObject(putRequest);
    }
}
```

File: weed/s3api/s3api_object_handlers.go [hunk 1]
```diff
@@ -115,6 +115,11 @@ func (s3a *S3ApiServer) PutObjectHandler(w http.ResponseWriter, r *http.Request)
 			dataReader = mimeDetect(r, dataReader)
 		}
 
+		ttl := r.Header.Get("x-amz-meta-ttl")
```
Comment (Dimonyga) [inline weed/s3api/s3api_object_handlers.go:118]: https://aws.amazon.com/ru/blogs/aws/amazon-s3-object-expiration/
x-amz-expiration ???

**Pull request closed without merging**